src/Nri/Ui/Button/V6.elm

```diff
@@ -0,0 +1,813 @@
+        [ Css.display Css.inlineBlock
+        , Css.backgroundRepeat Css.noRepeat
+        , Css.backgroundColor Css.transparent
+        , Css.backgroundPosition Css.center
+        , Css.backgroundSize Css.contain
+        , Css.border Css.zero
+        , Css.width (Css.px 15)
+        , Css.height (Css.px 15)
+        , Css.padding Css.zero
+        , Css.margin2 Css.zero (Css.px 6)
+        , Css.cursor Css.pointer
+        , Css.color Colors.azure
+        ]
+        [ Events.onClick config.onClick
+        , Attributes.type_ "button"
+        , -- reference: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_button_role#Labeling_buttons
+          Widget.label config.label
+        ]
+        [ Icon.icon { alt = "Delete", icon = Icon.xSvg assets } ]
+
+
+
+-- TOGGLE BUTTON
+
+
+{-| Buttons can be toggled into a pressed state and back again.
+-}
+type alias ToggleButtonConfig msg =
+    { label : String
+    , onSelect : msg
+    , onDeselect : msg
+    , pressed : Bool
+    }
+
+
+{-| -}
+toggleButton : ToggleButtonConfig msg -> Html msg
+toggleButton config =
+    let
+        toggledStyles =
+            if config.pressed then
+                Css.batch
+                    [ Css.color Colors.gray20
+                    , Css.backgroundColor Colors.glacier
+                    , Css.boxShadow5 Css.inset Css.zero (Css.px 3) Css.zero (ColorsExtra.withAlpha 0.2 Colors.gray20)
+                    , Css.border3 (Css.px 1) Css.solid Colors.azure
+                    , Css.fontWeight Css.bold
+                    ]
+
+            else
+                Css.batch
+                    []
+    in
+    Nri.Ui.styled Html.button
+        (styledName "toggleButton")
+        [ buttonStyles Medium WidthUnbounded SecondaryColors
+        , toggledStyles
+        ]
+        [ Events.onClick
+            (if config.pressed then
+                config.onDeselect
+
+             else
+                config.onSelect
+            )
+        , Widget.pressed <| Just config.pressed
+
+        -- reference: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_button_role#Labeling_buttons
+        , Role.button
+
+        -- Note: setting type: 'button' removes the default behavior of submit
+        -- equivalent to preventDefaultBehavior = false
+        -- https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#attr-name
+        , Attributes.type_ "button"
+        ]
+        [ viewLabel Nothing config.label ]
+
+
+{-| Inputs can be a clickable thing used in a form
+-}
+type alias InputConfig =
+    { content : Html Never
+    , name : String
+    , size : ButtonSize
+    , style : ButtonStyle
+    , value : String
+    }
+
+
+
+-- LINKS THAT LOOK LIKE BUTTONS
+
+
+{-| Links are clickable things with a url.
+
+NOTE: Links do not support two-line labels.
+
+-}
+type alias LinkConfig =
+    { label : String
+    , icon : Maybe IconType
+    , url : String
```

Comment:
The other thing I was wondering about was using `String` here for url - it seems like maybe we should pass in the wrapped Url type rather than a string. I know we do this in V5 as well, but I'm not quite sure why...

Reply:
Looks like it's always been `String`, but I agree that `Url` would be better!

Reply:
Yeah! Weirdly Nri.Button wrapped over V2 to make it a URL... I'm not sure why we never did that in the noredink-ui package. 🤷‍ Anyways, either way works fine!